vectorstores/cloudsql/loader/documentloader.go

```diff
@@ -0,0 +1,441 @@
+package loader
+
+import (
+	"context"
+	"database/sql"
+	"encoding/csv"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"golang.org/x/exp/slices"
+	"regexp"
+	"strings"
+
+	"github.com/averikitsch/langchaingo/schema"
+	"github.com/averikitsch/langchaingo/textsplitter"
+	"github.com/averikitsch/langchaingo/util/cloudsqlutil"
+	"github.com/jackc/pgtype"
+)
+
+const (
+	defaultMetadataJSONColumn = "langchain_metadata"
+	defaultSchemaName         = "public"
+)
+
+// Document represents a loaded document with content and metadata.
+type Document struct {
+	Content  string
+	Metadata map[string]interface{}
+}
+
+// Config holds the configuration for the DocumentLoader.
+type Config struct {
+	engine             cloudsqlutil.PostgresEngine
+	query              string
+	tableName          string
+	schemaName         string
+	contentColumns     []string
+	metadataColumns    []string
+	metadataJSONColumn string
+	format             string
+	formatter          func(map[string]interface{}, []string) string
+}
+
+// DocumentLoader is responsible for loading documents from a Postgres database.
+type DocumentLoader struct {
```

Comment:
We shouldn't need the double layer of abstraction. Let's use DocumentLoader to hold the config instead of defining both.